app/Models/Model.php

```diff
@@ -33,11 +34,13 @@ public function getForeignKey()
 
     public function getMacros()
     {
-        $macros = $this->macros ?? [];
-        $macros[] = 'realCount';
-        $macros[] = 'last';
+        static $baseMacros = [
+            'getWithHasMore',
+            'last',
+            'realCount',
+        ];
```

Comment:
could also have declared it has `protected const` and used `ReflectionClass::getConstants(ReflectionClassConstant::IS_PUBLIC)` since it's php8 now 🤔

Reply:
that's just adding even more hacks